delete_me_proxies_example/HumanReadableCleaningSteps.Rmd

```diff
@@ -0,0 +1,118 @@
+      # Create new units column: update with detection limit units when appropriate
+      resultUnits = checkBelowDetect(., DetectionQuantitationLimitMeasure.MeasureUnitCode, 
+                                     ResultMeasure.MeasureUnitCode),
+      # Create new remark column: if censored result, flag with <
+      remarkCd = checkBelowDetect(., "<", "NA")) %>% 
+    # Update for Above detection limit
+    mutate( 
+      # Create new resultVal column: contains result value or detection level 
+      #   if greater than comment is present
+      resultVal = checkAboveDetect(., DetectionQuantitationLimitMeasure.MeasureValue, resultVal),
+      # Create new units column: update with detection limit units when appropriate
+      resultUnits = checkAboveDetect(., DetectionQuantitationLimitMeasure.MeasureUnitCode, resultUnits),
+      # Populate remark code column with ">" for these results
+      remarkCd = checkAboveDetect(., ">", remarkCd))
+}
+
+# Actually clean the data and use the functions above
+
+wqp_data_clean <- wqp_data %>% 
+  
+  # Create FLAG for true missing results
+  mutate(flag_missingResult = (is.na(ResultMeasureValue) & 
+                                 is.na(DetectionQuantitationLimitMeasure.MeasureValue)) | 
+           grepl("*Not Reported|Not Reported|not reported", ResultDetectionConditionText) |
+           stringr::str_detect(tidyr::replace_na(ResultCommentText, ''), "not collected|not analyzed")) %>% 
+  
+  # Harmonize to generic fractions  
+  mutate(ResultSampleFractionText = 
+           ifelse(ResultSampleFractionText %in% c("Dissolved", "Filtered, lab", "Filtered, field"), 
+                  yes = "filtered",
+                  no = ifelse(ResultSampleFractionText %in% c("Total", "Total Recoverable", 
+                                                              "Total Recovrble", "Recoverable"), 
+                              yes = "unfiltered", 
+                              no = ifelse(ResultSampleFractionText %in% c("Suspended"), 
+                                          yes = "particulate",
+                                          no = ifelse(is.na(ResultSampleFractionText) | 
+                                                        ResultSampleFractionText %in% 
+                                                        c("NA", "None", "Pot. Dissolved", "Filterable", 
+                                                          "Non-filterable", "Non-Filterable (Particle)"),
+                                                      yes = "unknown",
+                                                      no = ResultSampleFractionText))))) %>%
+  # Create FLAG if fraction is unknown (thus ambiguous)
+  mutate(flag_fracAmbiguous = ResultSampleFractionText == "unknown") %>% 
+  
+  # Flag records w/ zero values and not enough information to resolve <-- meaning ResultMeasureValue 
+  #   was zero but DetectionQuantitationLimitMeasure.MeasureValue was also zero or missing. 
```

Comment:
One thing i like about this doc is that there is a pretty clear intro description about the script as a whole _and_ more detailed comments within the munging. 👍